src/rest_framework_jwt/authentication.py

```diff
@@ -5,7 +5,7 @@
 import jwt
 
 from django.contrib.auth import get_user_model
-from django.utils.encoding import smart_text
+from django.utils.encoding import smart_str
```

Comment:
Would you mind handling this via the `compat` module? That's where the rest of the compatibility code is located so we preserve backwards compatibility is as much as possible.